### PR TITLE
Set systemd as default for Raspbian 12

### PR DESCRIPTION
**Pull request by jaevans**
Fix for #9169. Needed for Raspberry Pis running bookworm.

https://perforce.atlassian.net/browse/PUP-11949

### CODE REVIEW

**Review by Adawntoremember — APPROVED** [review]

**Review by Adawntoremember — APPROVED** [review]

**Comment by joshcooper** [comment]
Thanks @jaevans! Could you rebase your PR on `main` and retarget your PR? We've changed how backports are handled since you originally filed this bug.

**Comment by jaevans** [comment]
I rebased on `main`, that brought in a bunch of 8.x stuff, was that the correct thing to do? What else needs to be done for this PR?

**Comment by joshcooper** [comment]
Hi @jaevans could you do do the following:

```
git checkout main
git pull
git checkout systemd-for-raspbian-7.x
git reset --hard upstream/main
git cherry-pick a911d1cd2eaf129a738cc6466eea5defa6968aa4
git push --force <remote> HEAD
```

Then change the `base` branch for this PR to `main`. Once the changes are merged to `main`, we will handle backporting to 7.x provided it is low-risk and doesn't break existing provider suitability checks.

**Comment by joshcooper** [comment]
Hi @jaevans it seems your pull request contains a merge commit `d4c9652`. We prefer that PRs don't contain merge commits, otherwise, it complicates the git history. Could you execute the following:

```
git checkout systemd-for-raspbian-7.x
git fetch upstream
git reset --hard upstream/main
git cherry-pick 2dc127f
git push -f origin HEAD
```

The above commands assume `upstream` is the puppetlabs remote and `origin` is your fork. You can see which is which by executing `git remote -v`

**Comment by jaevans** [comment]
Sorry for the delay, should be fixed now.

**Review by joshcooper — APPROVED** [review]

**Comment by Rocco83** [comment]
What are the blockers for this PR?
Seems to me that the failing tests are not related to this one.

File: spec/unit/provider/service/systemd_spec.rb [hunk 1]
```diff
@@ -171,6 +171,13 @@
     end
   end
 
+  it "should be the default provider on raspbian12" do
+    allow(Facter).to receive(:value).with(:osfamily).and_return(:debian)
+    allow(Facter).to receive(:value).with(:operatingsystem).and_return(:raspbian)
+    allow(Facter).to receive(:value).with(:operatingsystemmajrelease).and_return("12")
```
Comment (joshcooper) [inline spec/unit/provider/service/systemd_spec.rb:177]: Puppet 8 (main) uses structured facts so these stubs need to be updated. Something like:

```diff
-    allow(Facter).to receive(:value).with(:osfamily).and_return(:debian)
-    allow(Facter).to receive(:value).with(:operatingsystem).and_return(:raspbian)
-    allow(Facter).to receive(:value).with(:operatingsystemmajrelease).and_return("12")
+    allow(Facter).to receive(:value).with(:os.family).and_return(:debian)
+    allow(Facter).to receive(:value).with(:os.name).and_return(:raspbian)
+    allow(Facter).to receive(:value).with(:os.release.major).and_return("12")
```

Comment (Rocco83) [reply spec/unit/provider/service/systemd_spec.rb:177]: @jaevans will you take care or have you given up (for whatever reason)?

**Comment by joshcooper** [comment]
I've created a new PR in https://github.com/puppetlabs/puppet/pull/9477 and updated the specs to stub facter correctly. You'll remain the original author for the commit. Thanks for your contribution!

**Pull request closed without merging**